code/modules/lighting/light_source.dm

```diff
@@ -142,28 +145,59 @@
 /datum/light_source/proc/apply_lum()
 	applied = 1
 	if(istype(source_turf))
-		for(var/atom/movable/lighting_overlay/O in view(light_range, source_turf))
-			var/strength = light_power * falloff(O)
+		#if LIGHTING_RESOLUTION == 1
+		for(var/turf/T in dview(light_range, source_turf, INVISIBILITY_LIGHTING))
+			if(T.lighting_overlay)
+				var/strength = light_power * falloff(T.lighting_overlay)
+				if(!strength) //Don't add turfs that aren't affected to the affected turfs.
+					continue
+
+				effect_r[T.lighting_overlay] = round(lum_r * strength, LIGHTING_ROUND_VALUE)
+				effect_g[T.lighting_overlay] = round(lum_g * strength, LIGHTING_ROUND_VALUE)
+				effect_b[T.lighting_overlay] = round(lum_b * strength, LIGHTING_ROUND_VALUE)
 
-			effect_r[O] = lum_r * strength
-			effect_g[O] = lum_g * strength
-			effect_b[O] = lum_b * strength
+				T.lighting_overlay.update_lumcount(effect_r[T.lighting_overlay], effect_g[T.lighting_overlay], effect_b[T.lighting_overlay])
```

Comment:
Gonna change this to round again instead of getting it from the list, slightly faster because BYOND.